pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java

```diff
@@ -694,6 +716,88 @@ public void testManagedLedgerBookieClientStats() throws Exception {
         p2.close();
     }
 
+    @Test
+    public void testAuthMetrics() throws IOException, AuthenticationException {
+        SecretKey secretKey = AuthTokenUtils.createSecretKey(SignatureAlgorithm.HS256);
+
+        AuthenticationProviderToken provider = new AuthenticationProviderToken();
+
+        Properties properties = new Properties();
+        properties.setProperty("tokenSecretKey", AuthTokenUtils.encodeKeyBase64(secretKey));
+
+        ServiceConfiguration conf = new ServiceConfiguration();
+        conf.setProperties(properties);
+        provider.initialize(conf);
+
+        String authExceptionMessage = "";
+
+        try {
+            provider.authenticate(new AuthenticationDataSource() {
+            });
+            fail("Should have failed");
+        } catch (AuthenticationException e) {
+            // expected, no credential passed
+            authExceptionMessage = e.getMessage();
+        }
+
+        String token = AuthTokenUtils.createToken(secretKey, "subject", Optional.empty());
+
+        // Pulsar protocol auth
+        String subject = provider.authenticate(new AuthenticationDataSource() {
+            @Override
+            public boolean hasDataFromCommand() {
+                return true;
+            }
+
+            @Override
+            public String getCommandData() {
+                return token;
+            }
+        });
+
+        getPulsar().addPrometheusRawMetricsProvider(stream -> {
+            try {
+                StringWriter writer = new StringWriter();
+                Set<String> metricsName = new HashSet<>(Arrays.asList("pulsar_auth_success", "pulsar_auth_failures"));
+                TextFormat.write004(writer, CollectorRegistry.defaultRegistry.filteredMetricFamilySamples(metricsName));
+                stream.write(writer.toString());
+            } catch (IOException e) {
+                throw new RuntimeException(e.getMessage(), e.getCause());
+            }
+        });
```

Comment:
It's does not make sense here to copy the implementation code to the test. If the implementation changed, the test also can passed.